qiskit_optimization/algorithms/optimization_algorithm.py

```diff
@@ -196,6 +196,26 @@ def get_correlations(self) -> np.ndarray:
                         correlations[i, j] -= prob
         return correlations
 
+    def get_probabilities(self, status=None) -> Dict[str, float]:
+        """
+        Returns samples, and function values that are successful.
+
+        Returns:
+            A dictionary with sample names, and function values.
+        """
+        sample_results = {}
+        if status is None:
+            for sample in self.samples:
+                sample_values = f"{''.join(map(str, map(int, sample.x)))}"
+                sample_results.update({sample_values: sample.probability})
+        else:
+            for sample in self.samples:
+                sample_values = f"{''.join(map(str, map(int, sample.x)))}"
```

Comment:
`sample.x` is not always binary values. How do you distinguish `x=[1, 0, 1]`, `x=[10, 1]` and `x=[101]`?
I also recommend to use list comprehension instead of `map` two times.